core/miner_node.go

```
@@ -0,0 +1,721 @@
+	*ProposalMiner
+}
+
+type InvalidProposalMiner struct {
+	*BaseMiner
+}
+
+type UnSupportMiner struct {
+
+}
+
+func geneBaseIdentityOp(opType types.MinerType, targetMiner *types.Miner) baseIdentityOp {
+	if opType == types.MinerTypeVerify {
+		return &VerifyMiner{BaseMiner:&BaseMiner{}}
+	} else {
+		if targetMiner == nil {
+			return &NormalProposalMiner{ProposalMiner:&ProposalMiner{BaseMiner:&BaseMiner{}}}
+		}
+		switch targetMiner.Identity {
+			case types.MinerNormal:
+				return &NormalProposalMiner{ProposalMiner:&ProposalMiner{BaseMiner:&BaseMiner{}}}
+			case types.MinerGuard:
+				return &GuardProposalMiner{ProposalMiner:&ProposalMiner{BaseMiner:&BaseMiner{}}}
+			case types.MinerPool:
+				return &MinerPoolProposalMiner{ProposalMiner:&ProposalMiner{BaseMiner:&BaseMiner{}}}
+			case types.InValidMinerPool:
+				return &InvalidProposalMiner{BaseMiner:&BaseMiner{}}
+			default:
+				return &UnSupportMiner{}
+		}
+	}
+}
+
+func (n *NormalProposalMiner) processMinerOp(mop mOperation,targetMiner *types.Miner,op MinerOp) error {
+	switch op {
+	case StakedAddOp:
+		err := n.checkStakeAdd(mop,targetMiner)
+		if err != nil{
+			return err
+		}
+		err = n.processStakeAdd(mop,targetMiner,checkUpperBound)
+		if err!=nil{
+			return err
+		}
+	case StakeAbortOp:
+		err := n.checkStakeAbort(mop,targetMiner)
+		if err != nil{
+			return err
+		}
+		err = n.processStakeAbort(mop,targetMiner)
+		if err!=nil{
+			return err
+		}
+	case ApplyGuardOp:
+		err := n.checkApplyGuard(mop,targetMiner)
+		if err != nil{
+			return err
+		}
+		err = n.processApplyGuard(mop,targetMiner)
+		if err != nil{
+			return err
+		}
+	case VoteMinerPoolOp:
+		err := n.checkVoteMinerPool(mop)
+		if err != nil{
+			return err
+		}
+		err = n.processVoteMinerPool(mop,targetMiner)
+		if err != nil{
+			return err
+		}
+	case ReduceTicketOp:
+		n.processReduceTicket(mop,mop.Target(),mop.GetBaseOperation().subTicket)
+	case CancelGuardOp:
+		fmt.Errorf("normal proposal can not be operated")
```

should return the error